pkg/web/metricsHandler_test.go

```diff
@@ -31,77 +32,53 @@ func setupMetricsHandler(service MetricsServiceInterface) *httptest.Server {
 	return httptest.NewServer(router)
 }
 
-func TestMetricsEndpointShouldPassReceivedDataToMetricsService(t *testing.T) {
-	metric := mobile.Metric{
-		ClientTimestamp: "1234",
-		ClientId:        "client123",
-		Data: &mobile.MetricData{
-			App: &mobile.AppMetric{
-				ID:         "deadbeef",
-				SDKVersion: "1.2.3",
-				AppVersion: "27",
-			},
-			Device: &mobile.DeviceMetric{
-				Platform:        "android",
-				PlatformVersion: "19",
-			},
-		},
+func postMetric(t *testing.T, s *httptest.Server, metric *mobile.Metric) (*http.Response, error) {
+	var buffer *bytes.Buffer
+	buffer = new(bytes.Buffer)
+	err := json.NewEncoder(buffer).Encode(metric)
+	assert.Nil(t, err, "did not expect an error marshaling metric")
+
+	return postBuffer(t, s, buffer)
+}
+
+func postBuffer(t *testing.T, s *httptest.Server, buffer *bytes.Buffer) (res *http.Response, err error) {
+	if buffer != nil {
+		res, err = http.Post(s.URL+"/metrics", "application/json", buffer)
+	} else {
+		res, err = http.Post(s.URL+"/metrics", "application/json", nil)
 	}
+	assert.Nil(t, err, "did not expect an error posting metrics")
+	_, err = ioutil.ReadAll(res.Body)
```

Comment:
should you add a `defer res.Body.Close()` here?

Reply:
Ok, i see you're closing it after using the function

Reply:
I think it should be okay to defer close from this function as long as the rest of the tests don't want to use the contents of the body, I'll do this :+1: